base/sparse/sparsematrix.jl

```diff
@@ -70,9 +70,9 @@ julia> nnz(A)
 3
 ```
 """
-nnz(S::SparseMatrixCSC) = Int(S.colptr[S.n + 1]-1)
-countnz(S::SparseMatrixCSC) = countnz(S.nzval)
-count(S::SparseMatrixCSC) = count(S.nzval)
+nnz(S::SparseMatrixCSC)      = Int(S.colptr[S.n + 1] - 1)
+count(S::SparseMatrixCSC)    = count(S.nzval)
+count(f, S::SparseMatrixCSC) = count(f, S.nzval) + f(zero(eltype(S)))*(prod(size(S)) - nnz(S))
```

Comment:
What's this new method? Should go in a different commit if that's really intended.

Reply:
This method was missing. There is already a (dense) generic fallback here https://github.com/JuliaLang/julia/blob/5645fb212b90910b06c5a06f08c746fdd5b82f80/base/reduce.jl#L698-L704 but not a sparse version. The `count(pred, x)` method is suggested by the deprecation warning so I think it makes sense to add it in this PR.

Reply:
Got it. Might be worth a word in the commit message.